### DISCUSSION TITLE
This project grabbed my attention and ...

### DISCUSSION THREAD

**Discussion by dg2k** [espeak-ng/espeak-ng#1689] [General]
... as a Multi Platform software (namely .NET MAUI Apps) developer, using the latest Visual Studio 2022 Preview (17,6.0 Preview 2.0), I thought I will give it a spin if this project compiles for me.

**[Note: Amended this posting with an Update at the end]**

- Followed the [guide here](https://github.com/espeak-ng/espeak-ng/blob/master/docs/building.md)  and added the `pcaudiolib` as a folder into to the espeak-ng src folder.
- On compilation attempt, got a hint to add the **WiX v3 - Visual Studio 2022 Extension** [from here](https://marketplace.visualstudio.com/items?itemName=WixToolset.WixToolsetVisualStudio2022Extension).
- Next compilation fail hint was to install the very toolset needed for WiX installer which is currently [WiX Toolset v3.11.2](https://github.com/wixtoolset/wix3/releases/tag/wix3112rtm). Downloading the **wix311.exe** installer did the trick.
- With that, compiling the whole solution (of 4 projects) was a success!

That is with the latest Visual Studio IDE that Microsoft can offer today.

Now, I wonder where to start with this exciting field of **Text to Speech** technology. I am about to venture into developing for the Oromo (om) language. The good thing is that the writing of this language is phonetic based -- you write  precisely as the word sounds -- and in that sense, it lends itself to near perfect algorithm for text to phonemes. Oromo uses double vowels for longer utterance and double consonants for stressed syllables (as a non-linguist sorry if I am using the wrong tech terms here).  There are no exceptions for written forms as all are phonetically written (and no oddities like the English "enough"). With the adoption of the Latin script for writing, which happened only recently in 1991, you can say the Oromo language has the youngest orthography (that I know, anyway).

While C# is my forte, I can find my ways with C -- (after all C# descended from C and C++ pedigree). I own the first edition of  [The C Programming Language, by Kernighan & Ritchie](https://en.wikipedia.org/wiki/The_C_Programming_Language). Trying to get a link via search, I just landed on a free PDF of Second Edition, what a bliss,  so now also have an eBook version.

Can someone please give me a heads-up where I should start this project, and from which project folders etc ...
Is the starting point from constructing phonemes or what?
What are good refences to kick-start as a project for a new language as I tried to outline?
Can a new project like the one I intend to do from scratch take advantage of existing works, and if so which ones, please?

**UPDATE AFTER A FEW HOURS**
I did some cursory research on this repository and I have found the following files quite relevant to what I wanted to do for the Oromo language. Good news that someone already started it so good news for me.
- dictsource/om_rules
- dictsource/om_list
- espeak-ng-data/om_dict
- phsource/ph_oromo
- phsource/voc/dh_om

So I am not starting from scratch after all. So my question needs to modify how I can build on the top of this. Thank you,

Sorry for the tide of questions and hope only shows my enthusiasm.

Nice meeting you all!

Dan.

**Comment by jaacoppi** [comment]
Pick a task that interests you. There are a lot of bugs and refactoring waiting for volunteers.

You should start by reading the documentation in docs/.

Improving a single language usually requires only minimal programming. Improving the codebase for all languages does require you to know a bit of C.

I've heard that our current Oromo implementation isn't very good. You can start by using it to get a sense of how it feels to you.
